### PR TITLE
UIREQ-135: Use documented react-intl patterns instead of stripes.intl

### PR DESCRIPTION
**Pull request by MaksymDryha**
According to [UIREQ-135](https://issues.folio.org/browse/UIREQ-135), all strings should be internationalized using the react-intl pattern constructs outlined [here](https://github.com/folio-org/stripes/blob/master/doc/i18n.md)

### CODE REVIEW

File: src/ViewRequest.js [hunk 1]
```diff
@@ -340,27 +340,47 @@ class ViewRequest extends React.Component {
...
+                <KeyValue
+                  label={<FormattedMessage id="ui-requests.requestMeta.status" />}
+                  value={_.get(request, ['requestMeta', 'status'], '-')}
+                />
               </Col>
               <Col xs={3}>
-                <KeyValue label={formatMessage({ id: 'ui-requests.requestMeta.expirationDate' })} value={stripes.formatDate(_.get(request, ['requestMeta', 'requestExpirationDate'])) || '-'} />
+                <KeyValue
+                  label={<FormattedMessage id="ui-requests.requestMeta.expirationDate" />}
+                  value={stripes.formatDate(_.get(request, ['requestMeta', 'requestExpirationDate'])) || '-'}
```
Comment (Godlevskyi) [inline src/ViewRequest.js]: please use \<FormattedDate> from react-intl

File: src/CancelRequestDialog.js [hunk 2]
```diff
@@ -130,10 +140,14 @@ class CancelRequestDialog extends React.Component {
...
-          )}
+          label={
+            <FormattedMessage
+              id="ui-requests.cancel.additionalInfoLabel"
+              values={{
+                required: reason.requiresAdditionalInformation ? '*' : ' '
+              }}
+            />
+          }
           placeholder={
```
Comment (cherewaty) [inline src/CancelRequestDialog.js]: I think this pattern would work to be able to switch the `placeholder` away from `formatMessage`:

- In the render, before the `return`:
```
const additionalInfoPlaceholderKey = reason.requiresAdditionalInformation ?
      'ui-requests.cancel.additionalInfoPlaceholderRequired :
      'ui-requests.cancel.additionalInfoPlaceholderOptional';
```
- Then wrap the `TextArea`:
```
<FormattedMessage id={additionalInfoPlaceholderKey}>
  {placeholder => {
    <TextArea
      label={
        <FormattedMessage
          id="ui-requests.cancel.additionalInfoLabel"
          values={{
            required: reason.requiresAdditionalInformation ? '*' : ' '
          }}
        />
      }
      placeholder={placeholder}
      value={additionalInfo}
      onChange={this.onChangeAdditionalInfo}
    />
  )}
</FormattedMessage>
```

Comment (MaksymDryha) [reply src/CancelRequestDialog.js]: Thanks, @cherewaty! I fixed the code in accordance with your advice

File: src/CancelRequestDialog.js [hunk 3]
```diff
@@ -129,29 +144,33 @@ class CancelRequestDialog extends React.Component {
           onChange={this.onChangeNotify}
         />
         */}
-        <TextArea
-          label={formatMessage(
-            { id: 'ui-requests.cancel.additionalInfoLabel' },
-            { required: reason.requiresAdditionalInformation ? '*' : ' ' }
+        <FormattedMessage id={additionalInfoPlaceholder}>
+          {message => (
```
Comment (Godlevskyi) [inline src/CancelRequestDialog.js]: In such constructs pleaser use meaningful names. In current case it is not clear what is message.
lease rename it to placeholder

File: src/CancelRequestDialog.js [hunk 4]
```diff
@@ -98,14 +102,25 @@ class CancelRequestDialog extends React.Component {
...
+      : 'ui-requests.cancel.additionalInfoPlaceholderOptional';
 
     if (!request) return null;
 
     return (
       <Modal
-        label={formatMessage({ id: 'ui-requests.cancel.modalLabel' })}
+        label={<FormattedMessage id="ui-requests.cancel.modalLabel" />}
         open={this.props.open}
         onClose={this.props.onClose}
```
Comment (anastasiaHahanova) [inline src/CancelRequestDialog.js]: u can also destructure `open` and `onClose`

File: src/CancelRequestDialog.js [hunk 5]
```diff
@@ -129,29 +144,33 @@ class CancelRequestDialog extends React.Component {
...
-            reason.requiresAdditionalInformation ?
-              formatMessage({ id: 'ui-requests.cancel.additionalInfoPlaceholderRequired' }) :
-              formatMessage({ id: 'ui-requests.cancel.additionalInfoPlaceholderOptional' })
-          }
-          value={additionalInfo}
-          onChange={this.onChangeAdditionalInfo}
-        />
+        </FormattedMessage>
         <Layout className="textRight">
           <Button onClick={this.props.onClose}>
```
Comment (anastasiaHahanova) [inline src/CancelRequestDialog.js]: and here

File: src/RequestForm.js [hunk 6]
```diff
@@ -363,41 +370,53 @@ class RequestForm extends React.Component {
...
+              onClick={onCancel}
+              title={title}
+              aria-label={title}
+            >
+              <span style={{ fontSize: '30px', color: '#999', lineHeight: '18px' }}>&times;</span>
+            </Button>
+          )}
+        </FormattedMessage>
       </PaneMenu>;
     const addRequestLastMenu =
```
Comment (anastasiaHahanova) [inline src/RequestForm.js]: in this case it is better to wrap `<PaneMenu>` in round brackets ()

File: src/UserForm.js [hunk 7]
```diff
@@ -42,9 +43,64 @@ class UserForm extends React.Component {
...
+      >
+        <FormattedMessage id="ui-requests.actions.selectAddressType">
+          {(message) => <option value="">{message}</option>}
+        </FormattedMessage>
+        {deliveryLocations.map(location => {
+          const {
+            value,
+            label,
+          } = location;
+
```
Comment (anastasiaHahanova) [inline src/UserForm.js]:
```
{deliveryLocations.map(({ value, label}) => {
```

File: src/UserForm.js [hunk 8]
```diff
@@ -42,9 +43,64 @@ class UserForm extends React.Component {
...
+        fullWidth
+      >
+        <FormattedMessage id="ui-requests.actions.selectPickupSp">
+          {message => <option value="">{message}</option>}
+        </FormattedMessage>
+        {servicePoints.map(servicePoint => {
+          const {
+            id,
+            name,
+          } = servicePoint;
```
Comment (anastasiaHahanova) [inline src/UserForm.js]: and here

**Pull request merged**